Gems/Atom/RPI/Code/Source/RPI.Public/Buffer/Buffer.cpp

```diff
@@ -252,7 +253,16 @@ namespace AZ
         {
             if (m_streamFence)
             {
-                m_streamFence->WaitOnCpu();
+                auto deviceCount{RHI::RHISystemInterface::Get()->GetDeviceCount()};
```

Comment:
Is m_streamFence a multidevice fence and if so cant this funcitonality be pushed within it's WaitOnCpu?

Reply:
We intentionally removed the `WaitOnCpu()` in the `MultiDeviceFence` class because it really waits on all devices which in most cases is not what you want and we wanted to prevent developers from accidentally using that function when porting to the multi-device code once it gets into development. This case here is the only exception we found, so we implemented that functionality directly here. In all other cases the wait should happen on one specific device such as in the `AttachmentReadback` class for example.